Fix invalid percent formatting bug #172688620

james-wills-sf opened the pull request:
[pivotal ticket](https://www.pivotaltracker.com/n/projects/1405352/stories/172688620)

Previously, if a user entered 5.5.5 into the AMI percent field, the
formatter would auto-format it to be 5.5% because of how parseFloat()
behaves. This change makes it so that 5.5.5 will trigger a validation
error and will stop the page from saving successfully.

![- 2020-05-11 at 3 48 19 PM](https://user-images.githubusercontent.com/64036574/81619870-6615a880-939f-11ea-8cee-47911007c4d9.png)

james-wills-sf committed:
Fix invalid percent formatting bug #172688620

Previously, if a user entered 5.5.5 into the AMI percent field, the
formatter would auto-format it to be 5.5% because of how parseFloat()
behaves. This change makes it so that 5.5.5 will trigger a validation
error and will stop the page from saving successfully.

akegan commented:
actual review app: https://dahlia-lap-full-pr-353.herokuapp.com/

akegan reviewed (approved):
This change + QA looks good. Currency fields have this same issue. Wondering if it makes sense to fix in this PR too or open up a separate ticket.

james-wills-sf committed:
Fix the same behavior for currency fields

Mark ill-formatted currency fields cause validation errors as well, just
like percent fields do.

james-wills-sf commented:
@akegan it was a simple enough fix that I added it as a new commit to this PR, can you take another look?

![currencyform2](https://user-images.githubusercontent.com/64036574/81726045-9579f300-943b-11ea-9303-965d30d97d0a.gif)

akegan commented on spec/javascript/utils/form/validations.test.js:
```diff
@@ -160,15 +160,20 @@ describe('validate', () => {
-        expect(validate.isValidCurrency(VALIDATION_MSG)('$2000$')).toEqual(undefined)
-        expect(validate.isValidCurrency(VALIDATION_MSG)('$2,000,')).toEqual(undefined)
-        expect(validate.isValidCurrency(VALIDATION_MSG)('$')).toEqual(undefined)
+      // Maybe validation should be enhanced for this case?
```
not sure. Users are prevented from entering these values at least on Chrome

james-wills-sf replied:
@akegan yep it's across all browsers, (it happens in the `identity()` function in `Field.js`. I'll just update the comment to explain why we don't care about that case

Edit: I just removed the comment entirely and updated the regex to at least not allow things like `,`, `$,`, `,0.00`

james-wills-sf committed:
Handle additional comma edge cases

akegan reviewed (approved):
looks good and QA looks good too
